### PR TITLE
Docs: Add obtain method to ValueSource sample

### PR DESCRIPTION
**Pull request by hfhbd**
### Context
The sample code does not contain the main method: `obtain` which is needed to implement a ValueSource.

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java [hunk 1]
```diff
@@ -58,7 +58,10 @@
  *         this.execOperations = execOperations;
  *     }
  *
- *     // ...
+ *     {@literal @}Nullable
```
Comment (ov7a) [inline subprojects/core-api/src/main/java/org/gradle/api/provider/ValueSource.java:61]: 💅 Can you also add `@Override` here?

**Review by ov7a — COMMENTED** [review]
Thank you for you contribution!

**Review by ov7a — APPROVED** [review]

**Comment by hfhbd** [comment]
Well, looks like sign-off does not work with GitHub when editing a file, fail...

**Comment by hfhbd** [comment]
I guess it is just a bug of GitHub: The first commit/edit was correctly signed off, only later ones directly using the Web UI wasn't.

**Pull request merged**